### PR TITLE
Increase PerPage option to 100 when calling the User's Team List API

### LINKED ISSUE #3498: Support GitHub teams(30+) in the single sign-on.
**What would you like to be added**:

In the Single sign-on feature with Github, it would be great if we could set up permissions with Role-Based Access Control, even if a user belongs to many teams (30+).

**Why is this needed**:

I am configuring single sign-on for access to PipeCD's Project using Github Enterprise.

The problem I am encountering is that when I assign a role such as editor on PipeCD Project to a newly created team, I get the following error.
```
auth-handler: Unable to find user {"error": "user (<my name>) not found in any of the 30 project teams"}
```

According to the github api documentation, only 30 teams are returned by default.
https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user

Therefore, the current code does not seem to be able to retrieve more than 30 teams.
https://github.com/pipe-cd/pipecd/blob/5c1e5197f14b71d/pkg/oauth/github/github.go#L107
https://github.com/google/go-github/blob/3e8a7f0fbe0e4402f9ad/github/teams.go#L529-L531

### PR DESCRIPTION
**Pull request by misukuro**
**What this PR does / why we need it**:
In the GitHub Sign-on feature, if a user is on more than 30 teams, there is a problem with not getting all teams.
As a temporary fix, increase the PerPage value to 100 in the List API.

**Which issue(s) this PR fixes**:
Rel #3498

**Does this PR introduce a user-facing change?**:

```release-note
Fix bug that causes login error when user belongs to too many teams
```

### CODE REVIEW

File: pkg/oauth/github/github.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (c *OAuthClient) GetUser(ctx context.Context) (*model.User, error) {
 	if err != nil {
 		return nil, err
 	}
-	teams, _, err := c.Teams.ListUserTeams(ctx, nil)
+	teams, _, err := c.Teams.ListUserTeams(ctx, &github.ListOptions{PerPage: 100})
```
Comment (khanhtc1202) [inline pkg/oauth/github/github.go:107]: this `100` value should be defined at this file level as const to avoid magic number pattern.

Comment (misukuro) [reply pkg/oauth/github/github.go]: Thanks for the review.
I have corrected it.

**Comment by khanhtc1202** [comment]
bright and shiny 🙌 
/approve

**Pull request merged**